src/libraries/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -3466,22 +3476,22 @@ public static BigInteger RotateRight(BigInteger value, int rotateAmount)
                     zd[dstIndex] = (part >> smallShift) | carry;
                     carry = part << carryShift;
 
-                    dstIndex++;
-                    srcIndex++;
+                    dstIndex--;
+                    srcIndex--;
                 }
-                while (srcIndex < xd.Length);
+                while ((uint)srcIndex < (uint)xd.Length);
```

Comment:
I believe the JIT should be handling the more natural pattern of `(srcIndex >= 0) && (srcIndex < xd.Length)`

If it isn't, this likely deserves a comment particularly given that it's doing reverse iteration, which is less common.

Reply:
I added a comment.

```cs
                while ((uint)srcIndex < (uint)xd.Length); // is equivalent to (srcIndex >= 0 && srcIndex < xd.Length)
                while ((uint)dstIndex < (uint)zd.Length) // is equivalent to (dstIndex >= 0 && dstIndex < zd.Length)
```